app/models/project.rb

```diff
@@ -56,6 +69,24 @@ class Project < ActiveRecord::Base
     scope
   }
 
+  # Methods used to negotiate temporary abstract attribute `docker_build_method`
+  def docker_build_method
+    result = DOCKER_BUILD_METHODS.detect do |build_method|
+      build_method[:method] != 'samson' && send("#{build_method[:method]}?")
+    end
+
+    result&.[](:method) || 'samson'
+  end
```

Comment:
can map to method early to avoid hash access:

```Ruby
default = "samson"
DOCKER_BUILD_METHODS.map { |h| h.fetch(:method) }.detect { |m| m != default && send("#{build_method[:method]}?") } || default
```